### DISCUSSION TITLE
Doku fehler

### DISCUSSION THREAD

**Discussion by Mephisto090** [Zefau/ioBroker.jarvis#1933] [Q&A]
Hey @mcuiobroker,

irgendwas haut in deiner Doku nicht hin

https://mcuiobroker.gitbook.io/jarvis-infos/jarvis/besonderheiten-v3/module/jsontable/v3.x.x/parameter-format

`"format":"val => val = ('0'+new Date(val).getDay()).slice(-2)+'.'+('0'+new Date(val).getMonth()).slice(-2)+'.'+new Date(val).getFullYear()+' '+('0'+new Date(val).getHours()).slice(-2)+':'+('0'+new Date(val).getMinutes()).slice(-2)+':'+('0'+new Date(val).getSeconds()).slice(-2)"`

![image](https://user-images.githubusercontent.com/85995176/193442378-0995048b-ad37-4d26-9de8-a0530ac74a11.png)
![image](https://user-images.githubusercontent.com/85995176/193442448-5a00e8d5-812d-4857-83c0-dfd890eca7a0.png)

Das zweite Datum ist nur zum vergleichen gedacht. :)

Adapter Backitup

Wie kann ich mal kontakt zu dir mal aufnehmen? Habe einige Verbesserungsvorschläge und hätte ein paar Ideen bei Texten

**Comment by mcuiobroker** [comment]
Für BackitUp gibt ein extra format siehe Doku:
(liegt am Format von Backitup, dies ist kein internationales Datum-format)
https://mcuiobroker.gitbook.io/jarvis-infos/jarvis/besonderheiten-v3/module/jsontable/v3.x.x/parameter-format#backitup-datum-umsetzen

**Reply by Mephisto090** [reply]
Bei der Rohtabelle hängt doch ein TimeStamp am ende dran. Diese habe ich versucht in ein DateTime Format zu bringen

![image](https://user-images.githubusercontent.com/85995176/193442856-d36d64df-7b04-4b4e-abeb-bfe8fade6557.png)

**Reply by mcuiobroker** [reply]
Zeig mal die Einstellungen für die Spalten.

**Reply by mcuiobroker** [reply]
Ja du hast recht, Formel war falsch.

**Comment by mcuiobroker** [comment]
Deine Ideen kannst du gern hier posten. Ich werde dann versuchen sie umzusetzen. Alles hier hinein oder als Datei.

**Reply by Mephisto090** [reply]
Du gibst schön viel Quellcode für Copy&Paste (was ich für Leute die keine Ahnung haben echt gut finde. Lernt man zwar nichts kommt aber zum Ziel). Oft fehlt aber irgendwie eine Beschreibung für die Leute die sich doch mehr damit beschäftigen wollen.

https://mcuiobroker.gitbook.io/jarvis-infos/jarvis/system-effektprog./timetable

Hier musste ich erstmal die Links durchklicken um zu wissen, dass es um ein Saugrobo geht. Soll ich da mal eine kleine Einleitung zu machen?

**Reply by mcuiobroker** [reply]
Gerne. Ich übernehme es dann.

**Reply by Mephisto090** [reply]
In diesem timeTable Beispiel geht es um das Steuern eines Saug- oder Mähroboter an einzelne Wochentagen. Somit lassen sich einzelne Wochentage "An" und "Aus" schalten. Das Beispiel ist speziell für ein Mähroboter, einige Daten können bei Saugroboter weggelassen werden.

On: Einschaltzeit
Off: Endzeit
XXX min: Dauer in Minuten
Opt: Rasenkanten aktivieren
WK: Hat der Roboter an diesem Tag gearbeitet



Was sagst du dazu? Gerne kannst du noch etwas anpassen.
Kannst du nicht selbst ein GitDiscussion anlegen? So müssen wir nicht Zefaus zumüllen.

**Reply by mcuiobroker** [reply]
Versuch mal :
https://github.com/mcuiobroker/docuJarvis/discussions

**Reply by Mephisto090** [reply]
Ich finde auch deine Leiste links könnte besser sortiert sein. Ich suche mich da echt ein Wolf und finde selten was. Wäre es nicht besser anderes zu sortieren. Wie findest du die Idee?

Beispiel:

-- Module
--- JsonTable
---- v2
----- TR064 - Nur aktive Geräte
----- .....
---- v3
----- JsonTable erstellen
----- ....

--- HTML Table
---- v2
----- .....
---- v3
----- ....

**Reply by Mephisto090** [reply]
Ich zeihe mal um

**Comment by Zefau** [comment]
getDay müsste getDate sein oder? Und getMonth liefert einen Index, nicht die Zahl. Hier musst du +1 dazu addieren.

**Reply by mcuiobroker** [reply]
genau

**Reply by Mephisto090** [reply]
Also so:

`"format":"val => val = ('0'+new Date(val).getDate()).slice(-2)+'.'+('0'+new Date(val).getMonth()+1).slice(-2)+'.'+new Date(val).getFullYear()+' '+('0'+new Date(val).getHours()).slice(-2)+':'+('0'+new Date(val).getMinutes()).slice(-2)"`

**Reply by mcuiobroker** [reply]
Fast:
`new Date(val).getMonth()+1` -> `parseInt(new Date(val).getMonth())+1`

Doku wurde angepasst.

**Reply by Mephisto090** [reply]
Ich fuchse mich gerade in JS. Softwareprogrammierung ist kein Problem :(

**Reply by Mephisto090** [reply]
Ist das nicht auch noch falsch?

https://mcuiobroker.gitbook.io/jarvis-infos/jarvis/besonderheiten-v3/module/jsontable/v3.x.x/parameter-format#mit-vorangestellter-0

**Reply by mcuiobroker** [reply]
`[{"field":"threadTime","format":"val => val = ('0'+new Date(val).getDate()).slice(-2)+'.'+('0'+(parseInt(new Date(val).getMonth())+1)).slice(-2)+'.'+new Date(val).getFullYear()"}]
` sieht rictig aus?

**Reply by Mephisto090** [reply]
![image](https://user-images.githubusercontent.com/85995176/193445390-98d3c7c0-129e-4f60-8c75-ff26cf348c25.png)
![image](https://user-images.githubusercontent.com/85995176/193445405-1e60ed4c-d6b6-455e-ac52-8c77c0a15361.png)

**Reply by mcuiobroker** [reply]
![image](https://user-images.githubusercontent.com/84818934/193445474-92e1e7ed-6bfe-4cc3-bd14-e14010a2bf6a.png)

**Reply by Mephisto090** [reply]
sorry mein fehler

**Comment by Zefau** [comment]
@mcuiobroker ab `v3.1.0-alpha.53` kann nun `date()` genutzt werden. Also `val => date(val, 'dd.MM.yyyy')` oder `val => date(val, 'dd.MM.yyyy HH:mm:ss')`, etc.

**Reply by mcuiobroker** [reply]
Entspricht das dem formatDate?

**Reply by Zefau** [reply]
https://date-fns.org/v2.29.3/docs/format
